src/components/ContactsAdd.js

```diff
@@ -1,39 +1,129 @@
-import { useState } from "react"
-import { useNavigate } from "react-router-dom";
+import { useState } from "react";
+import { Navigate, useNavigate } from "react-router-dom";
 
-function ContactsAdd(props) {
+// import ContactInfoForm from "./contactInfoForm";
 
-  // setContacts and contacts must be passed as props
-  // to this component so new contacts can be added to the
-  // state
-  const { setContacts, contacts } = props
+const initalUser = {
+  firstName: "",
+  lastName: "",
+  street: "",
+  city: "",
+  email: "",
+  linkedIn: "",
+  twitter: "",
+};
 
-  //TODO: Implement controlled form
-  //send POST to json server on form submit
+function ContactsAdd({ contacts, setContacts }) {
+  const [newContact, setNewContact] = useState(initalUser);
+  const navigate = useNavigate();
+
+  function handleChange(e) {
+    const { id, value } = e.target;
+    setNewContact({ ...newContact, [id]: value });
+  }
```

Comment:
Wow, your ```handleChange```function is so condensed! I definitely need to implement that!!

Reply:
Yes agree, that is a lot better than switch.